src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -32,6 +33,91 @@ internal static partial class PathInternal
         };
 
         /// <summary>
+        /// Adds the extended path prefix (\\?\) if not already present.
+        /// </summary>
+        internal static string AddExtendedPathPrefix(string path)
+        {
+            if (IsExtended(path))
+                return path;
+
+            if (IsExtendedUnc(path))
+                return path.Insert(2, PathInternal.UncExtendedPrefixToInsert); // Given \\server\share in longpath becomes \\?\UNC\server\share  => UNCLongPathPrefix + path.SubString(2); => The actual command simply reduces the operation cost.
```

Comment:
Yeah, that's wrong. Fixing. Considering adding a test project for just the "common" bits- thoughts? Idea would be add a test project to the common directory that includes the common files and can test the internals that way.